### PR TITLE
Update to fc39

### PR DESCRIPTION
**Pull request by EdwardSro**
Prepare rdma-core for FC39

### CODE REVIEW

File: buildlib/pyverbs_functions.cmake [hunk 1]
```diff
@@ -8,7 +8,7 @@ function(build_module_from_cfiles PY_MODULE MODULE_NAME ALL_CFILES LINKER_FLAGS)
   string(REGEX REPLACE "\\.so$" "" SONAME "${MODULE_NAME}${CMAKE_PYTHON_SO_SUFFIX}")
   add_library(${SONAME} SHARED ${ALL_CFILES})
   set_target_properties(${SONAME} PROPERTIES
-    COMPILE_FLAGS "${CMAKE_C_FLAGS} -fPIC -fno-strict-aliasing -Wno-unused-function -Wno-redundant-decls -Wno-shadow -Wno-cast-function-type -Wno-implicit-fallthrough -Wno-unknown-warning -Wno-unknown-warning-option -Wno-deprecated-declarations ${NO_VAR_TRACKING_FLAGS}"
+    COMPILE_FLAGS "${CMAKE_C_FLAGS} -fPIC -fno-strict-aliasing -Wno-unused-function -Wno-redundant-decls -Wno-shadow -Wno-cast-function-type -Wno-implicit-fallthrough -Wno-deprecated-declarations ${NO_VAR_TRACKING_FLAGS}"
```
Comment (jgunthorpe) [inline buildlib/pyverbs_functions.cmake:11]: This is probably going to break something else.. I guess you need to test the flags for compatability.

**Comment by jgunthorpe** [comment]
The fedora image the cbuild creates has only cython 3.0.6, that needs to be fixed too
Also you need to address:

CMake Warning (dev) at CMakeLists.txt:218 (FIND_PACKAGE):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

**Comment by EdwardSro** [comment]
> The fedora image the cbuild creates has only cython 3.0.6, that needs to be fixed too 

What should be "fixed" here? Currently, cython 3 has a ([bug](https://github.com/cython/cython/issues/5860)) that prevents us from compiling pyverbs. That's why I disabled pyverbs on any cython3 < 3.0.9 (next release) 

> CMake Warning (dev) at CMakeLists.txt:218 (FIND_PACKAGE): Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules are removed. Run "cmake --help-policy CMP0148" for policy details. Use the cmake_policy command to set the policy and suppress this warning.
> 
> This warning is for project developers. Use -Wno-dev to suppress it.

I'll address that. FindPythonInterp and FindPythonLibs are replaced with FindPython since cmake 3.12 - should we support backward compatibility with older cmake?

**Comment by jgunthorpe** [comment]
You have to make the CI pass, so either cbuild has to install a fixed cython somehow, or the rpm build should not fail when it reaches the missing pyverbs.

File: buildlib/pyverbs_functions.cmake (hunk 1 above)
Comment (EdwardSro) [reply buildlib/pyverbs_functions.cmake:11]: I don't see the mentioned c11 warnings/notes now.
Anyway, I understand that it's not related to FC39 specifically, so I'm removing this commit from this PR.

**Comment by EdwardSro** [comment]
Updated PR to:

- Work-around cython3 bug (pyverbs should work now on any supported cython version)
- Address cmake warnings
- Build FC39 package successfully

**Comment by EdwardSro** [comment]
Updated PR to:

- Work-around cython3 bug (pyverbs should work now on any supported cython version)
- Address cmake warnings
- Build FC39 package successfully

**Pull request merged**